### ISSUE TITLE
Error synchronizing an Offline Editing project with a geometry updated (Only PostgreSQL)

### ISSUE THREAD

**Issue by gacarrillor** [qgis/QGIS#38851]
**Steps to reproduce**

1. You need 2 tables in PostgreSQL: 1 geographic (polygon) and 1 geometryless.The polygon layer has a FK. (See the DDL below)
```
CREATE EXTENSION postgis;

CREATE SCHEMA test;

CREATE TABLE test.domain_table (
	id int8 NOT NULL,
	CONSTRAINT domain_table_pkey PRIMARY KEY (id)
);

CREATE TABLE test.polygon_layer (
	id bigserial NOT NULL,
	domain_value int8 NOT NULL,
	geometria geometry(MULTIPOLYGON, 4326) NOT NULL,
	CONSTRAINT polygon_layer_pkey PRIMARY KEY (id)
);
ALTER TABLE test.polygon_layer ADD CONSTRAINT polygon_layer_domain_value_fkey FOREIGN KEY (domain_value) REFERENCES test.domain_table(id);
```
2. Load both tables into QGIS, create a record in `domain_table` and create a polygon in `polygon_layer` making sure its `domain_value` attribute points to the record created in `domain_table`.
3. Convert to offline project using Offline Editing.
4. Using the offline layers, move a node of the polygon and save edits.
5. Synchronize the changes.
6. You get these error messages:

`Offline Editing Plugin: ERROR: 1 geometries not changed. Provider errors: PostGIS error while changing geometry values: ERROR: invalid input syntax for type bigint: "NULL" `

`Layer polygon_layer: PostGIS error while changing geometry values: ERROR: invalid input syntax for type bigint: "NULL" `

**QGIS and OS versions**

QGIS version | 3.14.0-Pi | QGIS code revision | 9f7028fd23
-- | -- | -- | --
Compiled against Qt | 5.12.8 | Running against Qt | 5.12.8
Compiled against GDAL/OGR | 3.0.4 | Running against GDAL/OGR | 3.0.4
Compiled against GEOS | 3.8.0-CAPI-1.13.1 | Running against GEOS | 3.8.0-CAPI-1.13.1
Compiled against SQLite | 3.31.1 | Running against SQLite | 3.31.1
PostgreSQL Client Version | 12.2 (Ubuntu 12.2-4) | SpatiaLite Version | 4.3.0a
QWT Version | 6.1.4 | QScintilla2 Version | 2.11.2
Compiled against PROJ | 6.3.1 | Running against PROJ | Rel. 6.3.1, February 10th, 2020
OS Version | Ubuntu 20.04 LTS

**Notes**

  + I have tested the same scenario in GeoPackage and it works like a charm.
  + I have tested the issue with several project configurations: with only the polygon layer in the project, including the domain_table to the project, adding the qgis relation. None of them works as expected.
 + Tested on QGIS 3.14.0 and in QGIS master (rev. https://github.com/qgis/QGIS/commit/1a83d21912).

**Comment by uclaros** [comment]
Does it work correctly if you change the id from bigserial to serial?

**Comment by gacarrillor** [comment]
@uclaros, yes, it works when using `id serial NOT NULL` for the `polygon_layer`.

Unfortunately, I cannot change the table structure.

**Comment by Pedro-Murteira** [comment]
@gacarrillor I've been testing this issue on QGIS 3.22.3. and I can't reproduce it. Did you come across this issue more recently?

**Comment by gacarrillor** [comment]
@Pedro-Murteira, thanks for testing.

> @gacarrillor I've been testing this issue on QGIS 3.22.3. and I can't reproduce it. Did you come across this issue more recently?

Not really, I didn't test it again. Feel free to close it if you're confident of your test.

**Comment by uclaros** [comment]
@Pedro-Murteira did you follow the described steps and fail to replicate? It still fails for me on last week's master...

**Comment by Pedro-Murteira** [comment]
@uclaros I believe I did, I can be wrong tho, I only tested this on 3.22.3, didn't give it a shot on master. If you're being able to reproduce it, we can re-open.

**Comment by gioman** [comment]
Confirmed still valid also on 3.22.3.
